.circleci/config.yml

```diff
@@ -41,51 +52,121 @@ jobs:
       - restore_cache:
           name: Restore Rust cache
           keys:
-            - v5-cargo-pr-tests-{{ checksum "Cargo.toml" }}
-
-      - run:
-          name: Install Rust toolchain 1.72 (with clippy and rustfmt)
-          command: |
-            rustup toolchain install 1.72-x86_64-unknown-linux-gnu
-            rustup component add clippy --toolchain 1.72-x86_64-unknown-linux-gnu
-            rustup component add rustfmt --toolchain 1.72-x86_64-unknown-linux-gnu
-
-      - run:
-          name: Install nightly
-          command: rustup toolchain install nightly-2023-01-03-x86_64-unknown-linux-gnu
-
-      - run:
-          name: Install riscv target
-          command: rustup target add riscv32imac-unknown-none-elf --toolchain nightly-2023-01-03-x86_64-unknown-linux-gnu
-
-      - run:
-          name: Install stdlib
-          command: rustup component add rust-src --toolchain nightly-2023-01-03-x86_64-unknown-linux-gnu
+            - v5-cargo-pr-tests-{{ .Branch }}
 
       - run:
           name: Check without Halo2
           command: cargo check --all --no-default-features --profile pr-tests
 
       - run:
           name: Build
-          command: cargo build --all --all-features --profile pr-tests
+          command: cargo build --bins --tests --benches --all --all-features --profile pr-tests
 
       - save_cache:
           name: Save Rust cache
-          key: v5-cargo-pr-tests-{{ checksum "Cargo.toml" }}
+          key: v5-cargo-pr-tests-{{ .Branch }}
           paths:
             - ~/.cargo/registry
             - ~/.cargo/git
             - target
 
+      - persist_to_workspace:
+          root: .
+          paths:
+            - target/pr-tests
+            - pilcom
```

Comment:
we could also keep `.rustup` and `.cargo` so we don't need to reinstall them

Reply:
We should use a custom docker image.